CalibCalorimetry/HcalPlugins/src/HcalTextCalibrations.cc

```diff
@@ -36,49 +36,52 @@ HcalTextCalibrations::HcalTextCalibrations(const edm::ParameterSet& iConfig)
+      setWhatProduced(this, &HcalTextCalibrations::produceGains).setConsumes(mTokens[objectName]);
       findingRecord<HcalGainsRcd>();
     } else if (objectName == "GainWidths") {
-      setWhatProduced(this, &HcalTextCalibrations::produceGainWidths);
+      setWhatProduced(this, &HcalTextCalibrations::produceGainWidths).setConsumes(mTokens[objectName]);
       findingRecord<HcalGainWidthsRcd>();
     } else if (objectName == "QIEData") {
-      setWhatProduced(this, &HcalTextCalibrations::produceQIEData);
+      setWhatProduced(this, &HcalTextCalibrations::produceQIEData).setConsumes(mTokens[objectName]);
       findingRecord<HcalQIEDataRcd>();
     } else if (objectName == "QIETypes") {
-      setWhatProduced(this, &HcalTextCalibrations::produceQIETypes);
+      setWhatProduced(this, &HcalTextCalibrations::produceQIETypes).setConsumes(mTokens[objectName]);
       findingRecord<HcalQIETypesRcd>();
     } else if (objectName == "ChannelQuality") {
-      setWhatProduced(this, &HcalTextCalibrations::produceChannelQuality, edm::es::Label("withTopo"));
+      setWhatProduced(this, &HcalTextCalibrations::produceChannelQuality, edm::es::Label("withTopo"))
+          .setConsumes(mTokens[objectName]);
       findingRecord<HcalChannelQualityRcd>();
     } else if (objectName == "ZSThresholds") {
-      setWhatProduced(this, &HcalTextCalibrations::produceZSThresholds);
+      setWhatProduced(this, &HcalTextCalibrations::produceZSThresholds).setConsumes(mTokens[objectName]);
       findingRecord<HcalZSThresholdsRcd>();
     } else if (objectName == "RespCorrs") {
-      setWhatProduced(this, &HcalTextCalibrations::produceRespCorrs);
+      setWhatProduced(this, &HcalTextCalibrations::produceRespCorrs).setConsumes(mTokens[objectName]);
       findingRecord<HcalRespCorrsRcd>();
     } else if (objectName == "LUTCorrs") {
-      setWhatProduced(this, &HcalTextCalibrations::produceLUTCorrs);
+      setWhatProduced(this, &HcalTextCalibrations::produceLUTCorrs).setConsumes(mTokens[objectName]);
       findingRecord<HcalLUTCorrsRcd>();
     } else if (objectName == "PFCorrs") {
-      setWhatProduced(this, &HcalTextCalibrations::producePFCorrs);
+      setWhatProduced(this, &HcalTextCalibrations::producePFCorrs).setConsumes(mTokens[objectName]);
       findingRecord<HcalPFCorrsRcd>();
     } else if (objectName == "TimeCorrs") {
-      setWhatProduced(this, &HcalTextCalibrations::produceTimeCorrs);
+      setWhatProduced(this, &HcalTextCalibrations::produceTimeCorrs).setConsumes(mTokens[objectName]);
       findingRecord<HcalTimeCorrsRcd>();
     } else if (objectName == "L1TriggerObjects") {
-      setWhatProduced(this, &HcalTextCalibrations::produceL1TriggerObjects);
+      setWhatProduced(this, &HcalTextCalibrations::produceL1TriggerObjects).setConsumes(mTokens[objectName]);
       findingRecord<HcalL1TriggerObjectsRcd>();
     } else if (objectName == "ElectronicsMap") {
       setWhatProduced(this, &HcalTextCalibrations::produceElectronicsMap);
```

Comment:
Why are the `setConsumes` calls missing for some records? For example, `setConsumes` calls are present in very similar code for `HcalElectronicsMapRcd` in HcalHardcodeCalibrations.cc: https://github.com/cms-sw/cmssw/blob/65170ad24d1fabd8241aa0125cef8b2dbd0d0420/CalibCalorimetry/HcalPlugins/src/HcalHardcodeCalibrations.cc#L234

Reply:
Because the function in question doesn't consume anything

https://github.com/cms-sw/cmssw/blob/master/CalibCalorimetry/HcalPlugins/src/HcalTextCalibrations.cc#L290